src/cli.ts

```diff
@@ -23,18 +29,41 @@ async function main(argv: minimist.ParsedArgs) {
     process.exit(0)
   }
 
-  const argIn: string = argv._[0] || argv.input
-  const argOut: string = argv._[1] || argv.output
+  const argIn: string = argv._[0] || argv.input || argv.inDir
+  const argOut: string = argv._[1] || argv.output || argv.outDir
 
   try {
-    const schema: JSONSchema4 = JSON.parse(await readInput(argIn))
-    const ts = await compile(schema, argIn, argv as Partial<Options>)
-    await writeOutput(ts, argOut)
+    if (argv.inDir) {
+      const files = await glob(join(process.cwd(), argIn))
+      await processFiles(files, argOut, argv as Partial<Options>)
+    } else {
+      const schema: JSONSchema4 = JSON.parse(await readInput(argIn))
+      const ts = await compile(schema, argIn, argv as Partial<Options>)
+      await writeOutput(ts, argOut)
+    }
   } catch (e) {
     console.error(whiteBright.bgRedBright('error'), e)
     process.exit(1)
   }
+}
 
+function processFiles(files: string[], outDir: string, argv: Partial<Options>) {
+  return new Promise(async (res, rej) => {
```

Comment:
Could you use `Promise.all` here for better concurrency?

Reply:
I'll try this! Thank you for the recommendation